specs/select.spec.md

```diff
@@ -0,0 +1,156 @@
+
+---
+
+### API
+
+*Component Name*:
+- `fast-select`
+
+*Attributes:*
+- `autocomplete` - Allows the developer to provide a hint on how to search the content within the `<fast-option>`(s)
+- `autofocus` - automatically focuses the control
+- `disabled` - disables the control
+- `length` - The number of `<fast-option>` elements in the `<fast-select>` element
+- `multiple` - Allows user to select more than one `<fast-option>`
+- `name` - the name of the control
+- `options` - Returns a `HTMLOptionsCollection` of the `<fast-option>` elements contained by the `<fast-select>` element
+- `required` - boolean value that sets the field as required
+- `selectedIndex` - The index of the first or last selected `<fast-option>` element, depending on the value of `multiple`
+- `selectedOptions` - An `HTMLCollection` of the selected `<fast-option>` elements
+- `size` - If the `<fast-select>` is shown with a scrollbar, this represents how many `<fast-option>`s are visible.
+- `value` - string value of the number field, can be an empty string
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the `selectedIndex`'s value change.
+
+*Slots*
+- default - the list of option(s)
+- content - the content of the button
+- start - often times a glyph, icon, or button precedes the content
+- end - often times a glyph, icon, or button follows the content
+- expand-glyph - glyph that indicates the select is expandable, often represented by a downward chevron.
+
+### Anatomy and Appearance
+**Structure:**
+
+```html
+  <template>
+    <button
+          part="button"
+          class="button"
+    >
+        <slot name="start" part="start"></slot>
+        <slot name="content" part="content"></slot>
+        <slot name="end" part="end"></slot>
+        <slot name="expand-glyph" part="expand-glyph"></slot>
+    </button>
+    <ul
+        part="listbox"
+        class="listbox"
+    >
+        <slot></slot>
+    </ul>
+  </template>
+```
+
+## Implementation
+
+```html
+<fast-select>
+    <fast-option>Option 1</fast-option>
+    <fast-option>Option 2</fast-option>
+    <fast-option>Option 3</fast-option>
+    <fast-option>Option 4</fast-option>
+    <fast-option>Option 5</fast-option>
+</fast-select>
+```
+
+### States
+
+**open**: This state is applied to the `<fast-select>` when the listbox is visible to the user.
+
+**required**: An `<fast-option>` from the `<fast-select>` must be selected when the `required` attribute is set to `true`
+
+**valid**: The `<fast-select>` meets all its validation constraints, and is therefore considered to be valid.
+
+**invalid**: The `<fast-select>` does not meet its validation constraints, and is therefore considered to be invalid.
+
+### Accessibility
+
+*Select* are RTL compliant and support the following aria best practices for listbox [W3C aria-practices](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox)
```

Comment:
What are we using for a "listbox"?  We had that functionality wrapped in a component in the React version, should we do that again?  Listbox behavior shows up in other components like "auto  suggest" so we should be making it sharable somehow.

Reply:
This is a good point...in fact there is plenty of scenarios where list box is needed on its own and there's a lot in common between the components.

Reply:
The was the w3c has it in their spec is a `<ul>` with a role of `listbox` which is shown in the template.

![image](https://user-images.githubusercontent.com/37851214/89670367-f5930080-d895-11ea-93c1-be1f0e26764d.png)